### PR TITLE
Instrument `async_pools`

### PR DESCRIPTION
**Pull request by jacekwegr**
This PR updates the metrics in `async_pools` to use `mongoose_instrument`. Events are tested in separate suites due to the need for specific types of pools for testing. That's why one type of event is tested in `mod_inbox` and the other one in `mod_mam`.

Some of the tests do not check the exact number of events because the number of events can depend on how long a given operation takes. For example, `async_pool_flush.timed` can be executed a different number of times depending on how many messages are flushed in a given time interval.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/231865) / elasticsearch_and_cassandra_mnesia / 30bdbcde931e09d32af635bf8909a1f0028736dc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4337/231865/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4337/231865/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%404a9ad2705848.2024-07-25_14.09.03/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/232003) / elasticsearch_and_cassandra_mnesia / 60e1b6febacd220e3fd90ebe8d3ea1fcdd318215
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4337/232003/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4337/232003/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40b32128f89aa3.2024-07-25_14.29.14/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/232139) / elasticsearch_and_cassandra_mnesia / bd46278b2f58cdb3b29a732f04cf06e7128308e3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4337/232139/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4337/232139/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40562fd04e597e.2024-07-25_14.57.52/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/232578) / elasticsearch_and_cassandra_mnesia / 11f0193dedc8aa600cd7e4ebd018187632c6d41a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4337/232578/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4337/232578/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40ca7b40687575.2024-07-26_10.53.35/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[small_tests_25](https://circleci.com/gh/esl/MongooseIM/232703) / small_tests / 0f544a823ac3c98c3e16c7cac78e7a95116713b0
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4337/232703/small_tests.25.3.2.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4337/232703/small_tests.25.3.2.9-amd64/small/ct_run.mongooseim@localhost.2024-07-26_11.57.53/lib.mongooseim.logs/run.2024-07-26_11.57.53/suite.log.html)

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/232890) / elasticsearch_and_cassandra_mnesia / 3c21c8cca34528499f2a8409c3f4840b81b6b583
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4337/232890/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4337/232890/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40c3d80b2c3a52.2024-07-26_13.06.39/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

File: test/batches_SUITE.erl [hunk 1]
```diff
@@ -124,7 +121,7 @@ aggregation_might_produce_noop_requests(_) ->
     {ok, Server} = gen_server:start_link(?MODULE, [], []),
     Requestor = fun(1, _) -> timer:sleep(1), gen_server:send_request(Server, 1);
                    (_, _) -> drop end,
-    Opts = (default_aggregator_opts(Server))#{pool_id => ?FUNCTION_NAME,
+    Opts = (default_aggregator_opts(Server))#{pool_id => test_pool,
```
Comment (chrzaszcz) [inline test/batches_SUITE.erl:124]: I know the pools don't **need** different id's, but why change it?

**Review by chrzaszcz — APPROVED** [review]
Looks good 👍

File: test/batches_SUITE.erl (hunk 1 above)
Comment (jacekwegr) [reply test/batches_SUITE.erl:124]: It was causing errors in the test suite because the events were executed with an unregistered `pool_id` label

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/233637) / elasticsearch_and_cassandra_mnesia / 925915a437c1483b07e08da6cac24484d6941ae5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4337/233637/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4337/233637/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40ef4063863115.2024-07-30_08.31.19/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/233825) / elasticsearch_and_cassandra_mnesia / 72c3de42877886bdcedca91770c34f9ae483b2a4
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4337/233825/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4337/233825/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40ee6340d9392e.2024-07-30_11.25.33/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Review by chrzaszcz — APPROVED** [review]
Looks great now 👍

**Pull request merged**